src/exceptions.h

```diff
@@ -65,6 +65,11 @@ class FileNotGoodException : public BaseException {
 	FileNotGoodException(const std::string &s): BaseException(s) {}
 };
 
+class DatabaseException : public BaseException {
```

Comment:
I can, but then we should merge it before this commit to use it. Did you agree with this ? If yes i will do a PR in ~15 min